# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xorg-server/hotfix-sbus-gcc14.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xorg-server-21.1.12/hw/xfree86/common/xf86sbusBus.c.vanilla	2024-07-23 12:23:52.111676144 +0200
+++ xorg-server-21.1.12/hw/xfree86/common/xf86sbusBus.c	2024-07-23 12:26:06.832663163 +0200
@@ -748,10 +748,10 @@
         sparcPromClose();
     }
     if (promPath) {
-        XNFasprintf(&GDev->busID, "SBUS:%s", promPath);
+        XNFasprintf((char**)&GDev->busID, "SBUS:%s", promPath);
         free(promPath);
     }
     else {
-        XNFasprintf(&GDev->busID, "SBUS:fb%d", sBus->fbNum);
+        XNFasprintf((char**)&GDev->busID, "SBUS:fb%d", sBus->fbNum);
     }
 }
